### PR TITLE
Limit RocksDB write buffer

### PR DESCRIPTION
**Pull request by Nashatyrev**
## PR Description

RocksDB `memTable` (aka write buffer) size grows with time. I observed ~500M size after a couple of hours running

https://github.com/facebook/rocksdb/wiki/MemTable

Limit this size with the same value as blockCache size for now. If required later we may add a separate option

Was observing `memTable` size with this tweak: https://github.com/Nashatyrev/artemis/commit/81d614964a925975395970bd0aadeb2ae2239900#diff-c9a5225b916a8eb579008f07e0d1784dR65-L142

Didn't found any drastic write performance degradation. Though not 100% sure about this 

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

File: storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/core/RocksDbInstanceFactory.java [hunk 1]
```diff
@@ -112,6 +112,7 @@ private static DBOptions createDBOptions(final RocksDbConfiguration configuratio
         .setBytesPerSync(1048576L)
         .setWalBytesPerSync(1048576L)
         .setMaxBackgroundFlushes(2)
+        .setDbWriteBufferSize(configuration.getCacheCapacity())
```
Comment (rolfyone) [inline storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/core/RocksDbInstanceFactory.java]: I did some playing just to see what happens if we make it higher. I think around 256MB has a good mix of the disk being used, might be a better starting point than 8MB...
I tried with 128MB and there were constant disk flushes happening, 256MB means the WAL buffers fit inside the db write buffer size, and its happier by the looks.
I could definitely see this being a configurable option for tweaking, i think sync will be when you'd potentially want more buffers so the sync can progress faster...

give `.setDbWriteBufferSize(268435456L)` a go...

Comment (Nashatyrev) [reply storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/core/RocksDbInstanceFactory.java]: @rolfyone thanks for this investigation! Did it 256M by default

**Review by ajsutton — APPROVED** [review]
LGTM.  Merge is going to be a bit messy I suspect...

**Comment by ajsutton** [comment]
Ah no, merge was simple on this side because I haven't merged the metadata change yet.  So the merge will be messy over there. :) But I'll be able to sort it out.

**Pull request merged**